paimon-core/src/main/java/org/apache/paimon/utils/SnapshotManager.java

```diff
@@ -222,6 +223,46 @@ public Optional<Snapshot> latestSnapshotOfUser(String user) {
         return Optional.empty();
     }
 
+    /**
+     * Traversal snapshots from latest to earliest safely, this is applied on the writer side
+     * because the committer may delete obsolete snapshots, which may cause the writer to encounter
+     * unreadable snapshots.
+     */
+    public void traversalSnapshotsFromLatestSafely(Function<Snapshot, Boolean> consumer) {
```

Comment:
How about using `java.util.function.Predicate` ?

Reply:
It should not be a predicate, it is a consumer, and just return a early break boolean.

Reply:
Ok, make sense.